internal/print/error.go

```diff
@@ -0,0 +1,31 @@
+package print
+
+import (
+	"fmt"
+	"github.com/signadot/cli/internal/config"
+	"io"
+)
+
+func printErrorJson(err error) error {
+
+	return nil
+}
+
+func Error(out io.Writer, err error, outputFormat config.OutputFormat) error {
```

Comment:
Looks like this will change the exit status based on the output format, which is odd.  

Can we keep the CLI exit status non zero on error?  Perhaps using [`errors.Join`](https://pkg.go.dev/errors#Join) when json or yaml or the default switch case, combining the argument error with the error result of the encoding would not change the exit code.

Reply:
<img width="850" alt="Screenshot 2024-02-14 at 12 08 48 PM" src="https://github.com/signadot/cli/assets/34346877/ed5130b5-73b9-4d53-8768-4b4d0ca74972">

WDYT on this approach? 
<img width="740" alt="Screenshot 2024-02-14 at 12 11 48 PM" src="https://github.com/signadot/cli/assets/34346877/743efcf0-9a20-4bff-8ca5-e59e279232a1">